src/CodeGen_Hexagon.cpp

```diff
@@ -40,6 +40,7 @@ using namespace llvm;
 #endif
 
 #define IPICK(is_128B, i64) (is_128B ? i64##_128B : i64)
+#define V62PICK(is_128B, is_v62, i64, v62_i64) (is_v62 ? IPICK(is_128B, v62_i64) : IPICK(is_128B, i64))
```

Comment:
I'm wondering if it would be better to not create a macro for specifically selecting the intrinsic given V62 and vector width as it would be better to expose the selection logic so it is more understandable and controllable (esp. as more architectures and vector width combinations are added).

For example, with this change, the broadcast is selected as:
 
 ```
        // Broadcasts
+        { V62PICK(is_128B, is_v62, Intrinsic::hexagon_V6_lvsplatw, Intrinsic::hexagon_V6_lvsplatb), u8v1,  "splat.b", {u8} },
+        { V62PICK(is_128B, is_v62, Intrinsic::hexagon_V6_lvsplatw, Intrinsic::hexagon_V6_lvsplath), u16v1,  "splat.h", {u16} },
         { IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatw), u32v1,  "splat.w", {u32} },
```

but it could be more clearly selected without the macro as:

 ```
        // Broadcasts
        { is_v62 ?
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatb) :
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatw), u8v1,  "splat.b", {u8} },
        { is_v62 ?
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplath) :
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatw), u16v1,  "splat.h", {u16} },
        { IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatw), u32v1, "splat.w", {u32} }, 
```

...or if people really wanted a macro instead of the ternary op, maybe just define a generic TGTPICK macro that doesn't specify V62 in the name (as it could be used for any feature) and doesn't include the vector width:

#define TGTPICK(feature, has, has_not) (feature ? has : has_not)

 ```
        // Broadcasts
        { TGTPICK(is_v62,
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatb),
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatw)), u8v1,  "splat.b", {u8} },
        { TGTPICK(is_v62,
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplath),
            IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatw)), u16v1,  "splat.h", {u16} },
        { IPICK(is_128B, Intrinsic::hexagon_V6_lvsplatw), u32v1, "splat.w", {u32} }, 
```